netci.groovy

```diff
@@ -2199,6 +2359,51 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
                     // Basic archiving of the build, no pal tests
                     Utilities.addArchival(newJob, "bin/Product/**,bin/obj/*/tests/**/*.dylib,bin/obj/*/tests/**/*.so", "bin/Product/**/.nuget/**")
                     break
+                case 'arm':
+                    // Non-Windows ARM cross builds on hardware run on Ubuntu only
+                    assert (os == 'Ubuntu')
+
+                    // Debugging:
+                    // 1. show the IP address of the machine we're running on.
+                    // 2. show the directory the script is run in.
+                    // 3. show what's in this directory (currently).
+                    // 4. how much disk space is there?
+                    // etc.
+                    // use "|| true" to ignore all error codes
+                    buildCommands += """\
+uname -a || true
+ifconfig || true
+pwd || true
+ls -aF || true
+df -H || true
+printenv || true
+"""
+
+                    // Cross build the Ubuntu/arm product using docker with a docker image that contains the correct
+                    // Ubuntu cross-compilation toolset (running on a Ubuntu x64 host).
+
+                    def dockerImage = getDockerImageName(architecture, os, true)
+                    def dockerCmd = "docker run -i --rm -v \${WORKSPACE}:\${WORKSPACE} -w \${WORKSPACE} -e ROOTFS_DIR=/crossrootfs/arm ${dockerImage} "
+
+                    buildCommands += "${dockerCmd}\${WORKSPACE}/build.sh ${lowerConfiguration} ${architecture} cross"
+
+                    // Then, using the same docker image, generate the CORE_ROOT layout using build-test.sh to
+                    // download the appropriate CoreFX packages.
+
+                    buildCommands += "${dockerCmd}\${WORKSPACE}/build-test.sh ${lowerConfiguration} ${architecture} cross generatelayoutonly"
```

Comment:
We do not need docker for build-test. I will leave it up to you whether you keep it as it is or not, I have no preference. If we do keep it it would probably be nice to add a comment saying docker isn't entirely necessary.

Reply:
I'll leave it as is for consistency with the "build.sh" case, but add a comment.